modules/oe_theme_helper/tests/src/Functional/RetinaScaleEffectTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme_helper\Functional;
+
+use Drupal\FunctionalTests\Image\ToolkitTestBase;
+
+/**
+ * Tests that the Retina Scale effect upscales images appropriately.
+ *
+ * @group image
+ */
+class RetinaScaleEffectTest extends ToolkitTestBase {
```

Comment:
You are only testing the multiplier. What if the image is is bigger than the config. Or if the upscale is checked?

Reply:
Tests added